src/s3/util/fileViewerManager.ts

```diff
@@ -2,30 +2,237 @@
 
-    public constructor(window: Window = Window.vscode(), outputChannel = ext.outputChannel) {
-        //this.cache = new Set<S3FileNode>()
+    public constructor(
+        cacheArn: Set<string> = new Set<string>(),
+        window: typeof vscode.window = vscode.window,
+        commands = Commands.vscode(),
+        tempLocation?: string
+    ) {
+        this.cacheArns = cacheArn
         //this.activeTabs = new Set<S3Tab>()
-        //this.window = window
-        this.outputChannel = outputChannel
-        //this.createTemp()
-        showOutputMessage('initializing manager', outputChannel)
+        this.window = window
+        this.commands = commands
+        this.tempLocation = tempLocation
+        this.outputChannel = ext.outputChannel
     }
 
     public async openTab(fileNode: S3FileNode): Promise<void> {
-        showOutputMessage(`     manager initialized, file: ${fileNode.file.key}`, this.outputChannel)
+        getLogger().debug(`++++++++++++++++++++++++++++++++manager was initialized, file: ${fileNode.file.key}`)
+        showOutputMessage(
+            `++++++++++++++++++++++++++++++++manager was initialized, file: ${fileNode.file.key}`,
+            this.outputChannel
+        )
+
+        const fileLocation = await this.getFile(fileNode)
+        if (!fileLocation) {
+            return
+        }
+        getLogger().debug(`file to be opened is: ${fileLocation}`)
+        showOutputMessage(`file to be opened is: ${fileLocation}`, this.outputChannel)
+        //TODOD:: delegate this logic to S3Tab.ts
+        //this will display the document at the end
+        this.window.showTextDocument(fileLocation)
+    }
+
+    /**
+     * Fetches a file from S3 or gets it from the local cache if possible.
+     */
+    public async getFile(fileNode: S3FileNode): Promise<vscode.Uri | undefined> {
+        if (!this.tempLocation) {
+            await this.createTemp()
+        }
+        const targetPath = await this.createTargetPath(fileNode)
+        const targetLocation = vscode.Uri.file(targetPath)
+
+        const tempFile: vscode.Uri | undefined = await this.getFromTemp(fileNode)
+        //if it was found in temp, return the Uri location
+        if (tempFile) {
+            return tempFile
+        }
+
+        //needs to be downloaded from S3
+        //if file is +4MB, prompt user to confirm before proceeding
+        //const uri = vscode.Uri.file('')
+
+        if (fileNode.file.sizeBytes === undefined) {
+            const message = localize(
+                'AWS.s3.fileViewer.warning.noSize',
+                "File size couldn't be determined. Continue with download?"
+            )
+            const args = {
+                prompt: message,
+                confirm: localize('AWS.generic.continueDownload', 'Continue with download'),
+                cancel: localize('AWS.generic.cancel', 'Cancel'),
+            }
+
+            if (!(await showConfirmationMessage(args, this.window))) return undefined
```

Comment:
Unsure if we have a prettier rule for this, but all `if` statements should be on separate lines with curly braces. Goes for other places within this file.

Reply:
Prettier won't do this but I'm sure there's an ESLint rule.